### PR TITLE
Feat/api implementation

### PR DESCRIPTION
**Pull request by MuhammadArham-Traversaal**

### CODE REVIEW

File: utils/env_vars.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    secrets = get_secret_value_response['SecretString']
+    secrets = json.loads(secrets)
+    return secrets
+
+secrets = get_aws_secrets()
+
+def _get_env_var(key: str):
+    if key in secrets:
+        return secrets[key]
+    raise ValueError(f"AWS Secrets key missing: {key}")
```
Comment (vcalderon2009) [inline utils/env_vars.py:44]: I'd rephrase this to `Environment variable` rather than `AWS Secrets` since not all of them are `Secrets` at the moment.

File: utils/s3_utils.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            content_type = response.get('ContentType')
+            if content_type:
+                if 'word' in content_type:
+                    file_format = 'docx'
+                elif 'pdf' in content_type:
+                    file_format = 'pdf'
+                elif 'text' in content_type:
+                    file_format = 'txt'
+                else:
+                    file_format = 'unknown'
```
Comment (vcalderon2009) [inline utils/s3_utils.py:48]: How are you handling this type of format? Is this valid document type?

File: utils/env_vars.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+def get_aws_secrets():
+    if "AWS_REGION" not in os.environ:
+        raise ValueError("Environment Variable AWS_REGION unspecified. Can not access AWS Secrets.")
+    if "AWS_SECRETS_NAME" not in os.environ:
+        raise ValueError("Environment Variable AWS_SECRETS_NAME unspecified. Can not acess AWS Secrets.")
+    if "AWS_ACCESS_KEY_ID" not in os.environ:
+        raise ValueError("Environment Variable AWS_ACCESS_KEY_ID unspecified. Can not acess AWS Secrets.")
+    if "AWS_SECRET_ACCESS_KEY" not in os.environ:
+        raise ValueError("Environment Variable AWS_SECRET_ACCESS_KEY unspecified. Can not acess AWS Secrets.")
```
Comment (vcalderon2009) [inline utils/env_vars.py:13]: Maybe you can condense this into a single for loop over each environment variable:

```suggestion
    for env_name in ["AWS_REGION", "AWS_SECRETS_NAME", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]:
        if os.environ.get(env_name) is None:
            raise ValueError(f"Environment Variable {env_name} unspecified. Can not access AWS Secrets.")
```

**Review by vcalderon2009 — CHANGES_REQUESTED** [review]
Most of it looks good. I added some observations, but looks good.
Also, have you thought of adding unit tests for some of the modules?

File: utils/s3_utils.py (hunk 2 above)
Comment (MuhammadArham-Traversaal) [reply utils/s3_utils.py:48]: This was existing code in the Huggingface Spaces for ProEmbed API. In the embedding model, we raise a File Type Not Supported Error if the file is not specified. Will need to confirm this from the team about what file types are supported and what we need to handle.
